### PR TITLE
Feature/call dotnet fn from js

### PR DESCRIPTION
**Pull request by 5380harshjain**
Made changes in the js generic method.
These changes will allow us to call addEventListener using the generic method.

### CODE REVIEW

File: BlazorApp1/Pages/Index.razor [hunk 1]
```diff
@@ -1,726 +1,102 @@
...
+    }
+
+    // this is static method which we have added in the event using dotnet
+
+    [JSInvokable("ChangeData")]
+    public static async Task ChangeData()
+    {
+        // Call the non-static method from the instance
+        await _instance.NonStaticMethod();
+        Console.WriteLine(_instance);
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:35]: need to remove

File: BlazorApp1/Pages/Index.razor [hunk 2]
```diff
@@ -1,726 +1,102 @@
...
-        /* ------------------------ LISTENERS -------------------------*/
-        myChartConfig.type = "column3D";
-        myChartConfig.renderAt = "chartContainer2";
-        myChartConfig.id = "CHART_ID";
-
-        myEvent.dataPlotClick = "function() { console.log('dataPlotClick')}";
-        myEvent.initialized = "function() { console.log('initialized')}";
-        myEvent.beforeRender = "function() { console.log('beforeRender')}";
-        myEvent.rendered = "function() { console.log('rendered')}";
+        // here we are calling .net method from javascript
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:75]: need to remove

File: BlazorApp1/Pages/Index.razor [hunk 3]
```diff
@@ -1,726 +1,102 @@
...
-
-        dataset2.seriesname = "Los Angeles Topanga";
-        dataset2.data = new[] { value2, value2, value3, value4, value3, value6, value6 };
-
-        dataset3.seriesname = "Los Angeles Topanga";
-        dataset3.data = new[] { value4, value1, value4, value4, value4, value4, value3 };
-        dataset3.renderas = "line";
 
-        mytrendline = new[] { new { line = new[] { new { startvalue = 17022, color = "#6baa01", valueOnRight = 1, displayvalue = "Average" } } } };
+        //await fusionChartsService.CallFusionChartsFunction("setChartAttribute", "CHART_ID", "caption", "newcaption");
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:96]: need to remove

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 4]
```diff
@@ -1,33 +1,44 @@
...
+  if (typeof value === "string" && value.includes(FUNCTION_KEYWORD)) {
+    let functionParts = value.split("function()")
+    let functionBody = functionParts[1]
+    let parsedFn = new Function(functionBody);
+    return parsedFn;
+  }
+  // if the value is not function then return as is
+  return value;
+}
+// Responsible for rendering any chart//
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:14]: remove the end of the comment //

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 5]
```diff
@@ -1,33 +1,44 @@
...
+window.FusionCharts.renderChart = (chartConfiguration) => {
+  const configAsJSObject = JSON.parse(chartConfiguration, this.parseFunction);
+  const chart = new FusionCharts(configAsJSObject);
+  chart.render();
+};
+//Generic Method to call any fusion chart method exluding except above methods//
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:20]: remove the end of the comment //

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 6]
```diff
@@ -1,33 +1,44 @@
...
+window.FusionCharts.invokeChartFunction = (functionName, chartID, ...args) => {
+
+  var currentChart = FusionCharts(chartID);
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:23]: use const or let

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 7]
```diff
@@ -1,33 +1,44 @@
...
+
+  if (functionName === 'addEventListener' || (args.length > 0 && args[0][0] === "callback")) {
+
+    let event = args[0][1]
+    let functionAsString = args[0][2];
+    let callbackFn = parseFunction(null, functionAsString);
+    var result = currentChart[functionName].call(currentChart, event, callbackFn);
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:30]: avoid var, use let or const

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 8]
```diff
@@ -1,33 +1,44 @@
...
+
+    return String(result);
+  }
+
+  var result = currentChart[functionName].apply(currentChart, ...args);
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:35]: no var. let or const only

File: BlazorApp1/wwwroot/blazor-fusionCharts.js [hunk 9]
```diff
@@ -1,33 +1,44 @@
...
+  var typeofresult = typeof result;
```
Comment (meherhowji) [inline BlazorApp1/wwwroot/blazor-fusionCharts.js:36]: no var. let or const only

File: BlazorApp1/Pages/Index.razor [hunk 10]
```diff
@@ -1,726 +1,102 @@
...
-    {
-        await fusionChartsService.changeChartAttribute("CHART_ID", "caption", "New Chart Caption Changed");
-    }
-    private async Task obtainChartAttribute()
-    {
-        String axisName = await fusionChartsService.obtainChartAttribute("CHART_ID", "xAxisName");
     }
 }
-<SurveyPrompt Title="How is Blazor working for you?" />
+<SurveyPrompt Title="How is Blazor working for you?" />
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:102]: need to remove this

File: BlazorApp1/Pages/Index.razor [hunk 11]
```diff
@@ -1,726 +1,102 @@
...
-        myChart.theme = "fusion";
-
-        myCategories = new[] { new { catagory = new[] { new { label = "Day Start" } } } };
-
-        myEvent.initialized = "function() { updateTimeChartData() }";
         myDataSource.chart = myChart;
-        myDataSource.dataset = new[] { new { data = new[] { new { value = "35.3" } } } };
-        //myDataSource.categories = myCategories;
         myChartConfig.dataSource = myDataSource;
+        myChartConfig.id = "CHART_ID";
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:87]: meaningful chart id needed

File: BlazorApp1/FusionChartsInterop/FusionChartsService.cs [hunk 12]
```diff
@@ -1,6 +1,5 @@
 ﻿using Microsoft.AspNetCore.Components.RenderTree;
```
Comment (meherhowji) [inline BlazorApp1/FusionChartsInterop/FusionChartsService.cs:1]: need to upgrade this project to .NET7

File: BlazorApp1/Pages/Index.razor [hunk 13]
```diff
@@ -1,726 +1,102 @@
 ﻿@page "/"
 @using Microsoft.JSInterop;
-
+@inject IJSRuntime JSRuntime
 @inject FusionChartsService fusionChartsService;
-
 <PageTitle>Index</PageTitle>
 <h2>1. LARGE DATA</h2>
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:6]: meaningful title

File: BlazorApp1/Pages/Index.razor [hunk 14]
```diff
@@ -1,726 +1,102 @@
 ﻿@page "/"
 @using Microsoft.JSInterop;
-
+@inject IJSRuntime JSRuntime
 @inject FusionChartsService fusionChartsService;
-
 <PageTitle>Index</PageTitle>
```
Comment (meherhowji) [inline BlazorApp1/Pages/Index.razor:5]: meaningful title

File: BlazorApp1/Pages/Index.razor [hunk 15]
```diff
@@ -1,726 +1,102 @@
...
-<label for="pie3D">pie3D</label>
-<br />
-<button onclick="@destroyChart">Destroy Chart</button>
-<button onclick="@changeChartData">Change Chart Data</button>
-<button onclick="@obtainChartData">Get Chart Data</button>
-<button onclick="@changeChartAttribute">Change Chart Attribute (Caption) </button>
-<button onclick="@obtainChartAttribute">Get Chart Attribute (Width) </button>
+<!-- <button @onclick="ChangeData">Change Data</button> -->
+<!-- <p>Value: @value</p> -->
+Welcome to your new app.
```
[the remark this answers is not in the document]
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/Pages/Index.razor [hunk 16]
```diff
@@ -1,726 +1,102 @@
...
-<div>
-    <div id="chartContainer6"></div>
-</div>
-<h2>7. Stacked Column AND LINE</h2>
-<div>
-    <div id="chartContainer7"></div>
-</div>
+    public async Task NonStaticMethod()
+    {
+        Console.WriteLine("Hey! I'm not static method only");
```
[the remark this answers is not in the document]
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/Pages/Index.razor (hunk 2 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/Pages/Index.razor (hunk 11 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/FusionChartsInterop/FusionChartsService.cs (hunk 12 above)
Comment (5380harshjain) [reply BlazorApp1/FusionChartsInterop/FusionChartsService.cs:1]: done.

File: BlazorApp1/Pages/Index.razor (hunk 14 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: I've changed the title to Column2D Chart

File: BlazorApp1/Pages/Index.razor (hunk 1 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/Pages/Index.razor (hunk 10 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

File: BlazorApp1/Pages/Index.razor (hunk 13 above)
Comment (5380harshjain) [reply BlazorApp1/Pages/Index.razor]: done.

**Pull request closed without merging**